onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -409,12 +413,23 @@ def list(self, request, *args, **kwargs):
             xform_id, is_merged_dataset = qs[0] if qs else (lookup, False)
             pks = [xform_id]
             if is_merged_dataset:
-                xforms = MergedXForm.objects.filter(
-                    pk=xform_id, xforms__deleted_at__isnull=True)\
-                    .values_list('xforms', flat=True)
-                pks = [pk for pk in xforms if pk] or [xform_id]
+                merged_form = MergedXForm.objects.get(pk=xform_id)
+                pks = merged_form.xforms.filter(
+                    deleted_at__isnull=True).values_list('id', flat=True)
+                num_of_submissions = sum(
+                    merged_form.xforms.values_list(
+                        'num_of_submissions', flat=True))
```

Comment:
Should the above also filter by deleted_at is not null i.e.
```python
merged_form.xforms.filter(deleted_at__isnull=True).values_list('num_of_submissions', flat=True)
```
If the above is true, can the query in line 417 and 420 be merged into 1 to retrieve both the `id` and `num_of_submissions `?